blackjax/kernels.py

```diff
@@ -715,27 +721,116 @@ def run(rng_key: PRNGKey, position: PyTree):
             one_step_ = jax.jit(one_step)
 
         keys = jax.random.split(rng_key, num_steps)
-        last_state, warmup_chain = jax.lax.scan(
+        schedule = adaptation.window_adaptation.schedule(num_steps)
+        last_state, adaptation_chain = jax.lax.scan(
             one_step_,
-            (init_state, init_warmup_state),
+            (init_state, init_adaptation_state),
             (jnp.arange(num_steps), keys, schedule),
         )
+        last_chain_state, last_warmup_state, *_ = last_state
+
+        step_size, inverse_mass_matrix = final(last_warmup_state)
+        parameters = {
+            "step_size": step_size,
+            "inverse_mass_matrix": inverse_mass_matrix,
+            **extra_parameters,
+        }
+
+        def kernel(rng_key, state):
+            return step_fn(rng_key, state, logprob_fn, **parameters)
+
+        return AdaptationResults(last_chain_state, kernel, parameters)
+
+    return AdaptationAlgorithm(run)
+
+
+def pathfinder_adaptation(
```

Comment:
Could you move the pathfinder_adaptation to the bottom? it's easier to see the diff.

Reply:
Ok! We'll need another PR to re-arrange things in there. It's getting messy, hence #292